x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/util/filter_runtime_mappings.ts

```diff
@@ -68,17 +68,22 @@ function findFieldsInJob(job: Job, datafeed: Datafeed) {
 
   const aggs = datafeed.aggregations ?? datafeed.aggs;
   if (aggs !== undefined) {
-    findFieldsInAgg(aggs).forEach((f) => usedFields.add(f));
+    findFieldsInObject(aggs).forEach((f) => usedFields.add(f));
+  }
+
+  const query = datafeed.query;
```

Comment:
I tried creating a job which only used a runtime field in the query, but on job creation it seems to be removed from the datafeed config, so the job fails.

Datafeed config (before creating job):
```
{
  "query": {
    "term": {
      "regionrt": {
        "value": "eu-central-1_1",
        "boost": 1
      }
    }
  },
  "indices": [
    "cloudwatch-*"
  ],
  "runtime_mappings": {
    "regionrt": {
      "type": "keyword",
      "script": {
        "source": "emit(doc['region'].value + '_1')"
      }
    }
  },
  "scroll_size": 1000,
  "delayed_data_check_config": {
    "enabled": true
  },
  "job_id": "cloudwatch_search_rt2",
  "datafeed_id": "datafeed-cloudwatch_search_rt2"
}
```

Analysis config:
```
{
  "job_id": "cloudwatch_search_rt2",
  "description": "Uses a runtime field in the search only",
  "analysis_config": {
    "bucket_span": "1h",
    "detectors": [
      {
        "function": "mean",
        "field_name": "CPUUtilization",
        "partition_field_name": "instance",
        "detector_description": "mean(CPUUtilization) partitionfield=instance"
      }
    ],
    "influencers": [
      "instance",
      "region"
    ]
  },
  "analysis_limits": {
    "model_memory_limit": "11MB"
  },
  "data_description": {
    "time_field": "@timestamp",
    "time_format": "epoch_ms"
  },
  "model_plot_config": {
    "enabled": false,
    "annotations_enabled": false
  },
  "model_snapshot_retention_days": 10,
  "daily_model_snapshot_retention_after_days": 1,
  "allow_lazy_open": false,
  "groups": []
}
```

Creating a job with the same config as above, but with `regionrt` added as an influencer, it works correctly.

Reply:
fixed in ece974f5f2812d627c339ebced31e1129b41e420

Reply:
The job is now running correctly, but the cardinality check in the validation step seems to not return any documents:

![image](https://user-images.githubusercontent.com/7405507/111975404-78508f00-8af8-11eb-94da-340636129cfe.png)

Reply:
Fixed in https://github.com/elastic/kibana/pull/94760/commits/7f6a925ec2ebc94bb6ee9f1d6e7136cd40fff548